examples/cis2-wccd-upgrade/src/lib.rs

```diff
@@ -0,0 +1,2030 @@
+                    EntrypointName::new_unchecked("get_unpause_time"),
+                    Amount::zero(),
+                )?
+                .1;
+
+            let unpause_time = if let Some(mut unpause_time) = unpause_time {
+                unpause_time.get()?
+            } else {
+                return Err(concordium_cis2::Cis2Error::Custom(StateInvokeError));
+            };
+
+            ensure!(slot_time >= unpause_time, concordium_cis2::Cis2Error::Custom(ContractPaused));
+
+            let params: WrapParams = ctx.parameter_cursor().get()?;
+            // Get the sender who invoked this contract function.
+            let sender = ctx.sender();
+
+            let receive_address = params.to.address();
+
+            let (state, state_builder) = host.state_and_builder();
+
+            // Update the state.
+            state.mint(&TOKEN_ID_WCCD, amount.micro_ccd.into(), &receive_address, state_builder)?;
+
+            // Log the newly minted tokens.
+            logger.log(&Cis2Event::Mint(MintEvent {
+                token_id: TOKEN_ID_WCCD,
+                amount:   ContractTokenAmount::from(amount.micro_ccd),
+                owner:    sender,
+            }))?;
+
+            // Only log a transfer event if receiver is not the one who payed for this.
+            if sender != receive_address {
+                logger.log(&Cis2Event::Transfer(TransferEvent {
+                    token_id: TOKEN_ID_WCCD,
+                    amount:   ContractTokenAmount::from(amount.micro_ccd),
+                    from:     sender,
+                    to:       receive_address,
+                }))?;
+            }
+
+            // If the receiver is a contract: invoke the receive hook function.
+            if let Receiver::Contract(address, function) = params.to {
+                let parameter = OnReceivingCis2Params {
+                    token_id: TOKEN_ID_WCCD,
+                    amount:   ContractTokenAmount::from(amount.micro_ccd),
+                    from:     sender,
+                    data:     params.data,
+                };
+                host.invoke_contract(
+                    &address,
+                    &parameter,
+                    function.as_entrypoint_name(),
+                    Amount::zero(),
+                )
+                .unwrap_abort();
+                Ok(())
+            } else {
+                Ok(())
+            }
+        }
+    }
+}
+
+/// Unwrap an amount of wCCD tokens into CCD
+#[receive(
+    contract = "CIS2-wCCD",
+    name = "unwrap",
+    parameter = "UnwrapParams",
+    enable_logger,
+    mutable
+)]
+fn contract_unwrap<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    host: &mut impl HasHost<StateImplementation<S>, StateApiType = S>,
+    logger: &mut impl HasLogger,
+) -> ContractResult<()> {
+    let slot_time = ctx.metadata().slot_time();
+
+    match host.state().state_address {
+        None => bail!(concordium_cis2::Cis2Error::Custom(OnlyState)),
+
```

Comment:
As above, this might be easier to read  (i.e., less nesting) if you used

```rust
let state_address = host.state().state_address.ok_or(concordium_cis2::Cis2Error::Custom(OnlyState))?
...
```